### PR TITLE
GEOPY-1478: Deal with negative octree cell dimensions from legacy inversions

### PR DESCRIPTION
**Pull request by sebhmg**
**GEOPY-1478 - Deal with negative octree cell dimensions from legacy inversions**

### CODE REVIEW

**Comment by sebhmg** [comment]
see https://github.com/MiraGeoscience/simpeg-drivers/pull/22 to review linter fixes in a dedicated PR

File: simpeg_drivers/components/data.py [hunk 1]
```diff
@@ -131,9 +131,9 @@ def _initialize(self) -> None:
         if self.radar is not None and any(np.isnan(self.radar)):
             self.mask[np.isnan(self.radar)] = False
 
-        self.observed = self.filter(self.observed)
-        self.radar = self.filter(self.radar)
-        self.uncertainties = self.filter(self.uncertainties)
+        self.observed = self.filter(self.observed, self.mask)
+        self.radar = self.filter(self.radar, self.mask)
+        self.uncertainties = self.filter(self.uncertainties, self.mask)
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply simpeg_drivers/components/data.py:136]: I merged it in with Seb's issue that solved the environment.

[the remark this answers is not in the document]
Comment (sebhmg) [reply simpeg_drivers/components/data.py:136]: and I had merged in GEOPY-1224

[the remark this answers is not in the document]
Comment (sebhmg) [reply simpeg_drivers/components/data.py:136]: well, I am not sure where this is coming from anymore...

[the remark this answers is not in the document]
Comment (benk-mira) [reply simpeg_drivers/components/data.py:136]: I think Dom had a different way to solve the mypy error on one of his issues, I was hoping to merge it in, but there haven't been any changes to release/0.1.0.  I'll try and find it again..

[the remark this answers is not in the document]
Comment (sebhmg) [reply simpeg_drivers/components/data.py:136]: I referred to the wrong issue number: my branch had GEOPY-1477 merged in (https://github.com/MiraGeoscience/simpeg-drivers/pull/19), and some commits that @benk-mira  had applied already for linters. Possibly the one here can be reverted?

File: simpeg_drivers/components/meshes.py [hunk 2]
```diff
@@ -22,19 +22,37 @@
...
-    from geoh5py.workspace import Workspace
-
     from simpeg_drivers.components.data import InversionData
     from simpeg_drivers.components.topography import InversionTopography
 
 
+# TODO: Import this from newer octree-creation-app release
+def convert_octree_levels(mesh):
+    """
+    Convert octree cells indices and levels to CPP representation.
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply simpeg_drivers/components/meshes.py]: done

**Review by domfournier — APPROVED** [review]

**Pull request merged**